### PR TITLE
Update Kitchen/Travis tests

### PR DESCRIPTION
**Pull request by n-rodriguez**
### PR progress checklist (to be filled in by reviewers)

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [x] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Describe the changes you're proposing

This PR extends Kitchen/Inspec tests to all supported platforms.

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,20 +1,49 @@
+# -*- coding: utf-8 -*-
```
Comment (myii) [inline .travis.yml]: I know this isn't the complete file but can you base what you have got on the latest structure in the `template-formula`?

File: .travis.yml [hunk 2]
```diff
@@ -1,20 +1,49 @@
...
+    - INSTANCE: default-opensuse-leap-15-2018-3-py2
+    - INSTANCE: default-amazonlinux-2-2018-3-py2
+    - INSTANCE: default-arch-base-latest-2018-3-py2
+    - INSTANCE: default-debian-8-2017-7-py2
+    - INSTANCE: default-ubuntu-1604-2017-7-py2
+    - INSTANCE: default-centos-6-2017-7-py2
+    - INSTANCE: default-fedora-29-2017-7-py2
+    - INSTANCE: default-opensuse-leap-15-2017-7-py2
+    - INSTANCE: default-amazonlinux-2-2017-7-py2
+    - INSTANCE: default-arch-base-latest-2017-7-py2
```
Comment (myii) [inline .travis.yml]: Brilliant, everything is working!  Can we use the general-purpose matrix here?

```yaml
        platforms_matrix:
          # [os           , os_ver, salt_ver, py_ver,    inspec_suite]
          - [debian       ,  10   ,  develop,      3,         default]
          - [ubuntu       ,  18.04,   2019.2,      3,         default]
          - [amazonlinux  ,   2   ,   2019.2,      2,         default]
          - [arch-base    , latest,   2019.2,      2,         default]
          - [fedora       ,  29   ,   2018.3,      2,         default]
          - [opensuse/leap,  15   ,   2018.3,      2,         default]
          - [centos       ,   6   ,   2017.7,      2,         default]
```

File: kitchen.yml [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+  log_level: debug
+  salt_install: none
+  require_chef: false
+  formula: haproxy
+  salt_copy_filter:
+    - .kitchen
+    - .git
+  dependencies:
+    - name: hosts
+      path: test/salt/formulas
```
Comment (myii) [inline kitchen.yml]: Please moves this `dependencies` section below, to where indicated.

File: kitchen.yml [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+            - haproxy
+      pillars:
+        top.sls:
+          base:
+            '*':
+              - hosts
+              - haproxy
+      pillars_from_files:
+        haproxy.sls: test/salt/pillar/haproxy.sls
+        hosts.sls: test/salt/pillar/hosts.sls
```
Comment (myii) [inline kitchen.yml]: Would you mind switching these two around, to keep the consistent order of `hosts` and `haproxy` with above?

Also, the `dependencies` section to be moved underneath here, i.e.

```suggestion
        haproxy.sls: test/salt/pillar/haproxy.sls
      dependencies:
        - name: hosts
          path: test/salt/formulas
```

**Comment by n-rodriguez** [comment]
@daks feel free to submit your own PR

**Comment by daks** [comment]
@n-rodriguez why did you close this PR _(and other PR and issues elsewhere)_ when most of the work is already done?

**Pull request closed without merging**